src/classes/HiFiCommunicator.ts

```diff
@@ -214,56 +204,17 @@ export class HiFiCommunicator {
         }
 
         let mixerConnectionResponse;
+        let hostURLSafe;
+        
         try {
-            // TODO: Revisit this chunk of code later. We almost certainly don't want this to stay in the API code,
-            // but it's _very_ convenient for sample apps for right now.
-            let params = URLSearchParams && (typeof (location) !== 'undefined') && new URLSearchParams(location.search);
-            if (params && params.has("token") && (!hifiAuthJWT || hifiAuthJWT.length === 0)) {
-                hifiAuthJWT = params.get("token");
-            }
-
-            if (!hifiAuthJWT || hifiAuthJWT.length === 0) {
-                let errMsg = `Can't connect to API Server: No JWT.\nSee this guide for information about obtaining a JWT:\nhttps://www.highfidelity.com/api/guides/misc/getAJWT`;
-                return Promise.reject({
-                    success: false,
-                    error: errMsg
-                });
-            }
-
-            let webRTCSignalingAddress = "wss://loadbalancer-$STACKNAME.highfidelity.io:8001/?token=";
-            let isBrowserContext = typeof self !== 'undefined';
-            if (params && params.has("stack")) {
-                let url;
-                try {
-                    url = new URL(params.get("stack"));
-                } catch (e) { }
-
-                if (url) {
-                    webRTCSignalingAddress = url.href;
-                } else {
-                    webRTCSignalingAddress = webRTCSignalingAddress.replace('$STACKNAME', params.get("stack"));
-                }
-            } else if (isBrowserContext && window.location.hostname.indexOf("highfidelity.io") > -1) {
-                webRTCSignalingAddress = `wss://${window.location.hostname}:8001/?token=`;
-            } else if (isBrowserContext && BUILD_ENVIRONMENT && BUILD_ENVIRONMENT === "prod") {
-                webRTCSignalingAddress = `${HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT}/?token=`;
-            } else if (stackName) {
-                let url;
-                try {
-                    url = new URL(stackName);
-                } catch (e) { }
-
-                if (url) {
-                    webRTCSignalingAddress = url.href;
-                } else {
-                    webRTCSignalingAddress = webRTCSignalingAddress.replace('$STACKNAME', stackName);
-                }
-            } else if (!isBrowserContext) {
-                webRTCSignalingAddress = `${HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT}/?token=`;
-            } else {
-                webRTCSignalingAddress = webRTCSignalingAddress.replace('$STACKNAME', 'api-staging-02');
-            }
+            hostURLSafe = new URL(hostURL).hostname;
+        } catch(e) {
+            // If hostURL is not defined, we assign the default URL
+            hostURLSafe = hostURL ? hostURL : HiFiConstants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT;
```

Comment:
why wouldn't we want to do this:
```suggestion
            hostURLSafe = HiFi4Constants.DEFAULT_PROD_HIGH_FIDELITY_ENDPOINT;
```
?

Reply:
The first thing we do is trying to catch a complete URL (i.e. `https://api.highfidelity.com/`). If that fails we assume that either `hostURL` is undefined or it is not a complete URL (i.e `api.highfidelity.com`). Keep in mind that `new URL("api.highfidelity.com")` will fail.

Reply:
aha! okay, thanks for the explanation.

Reply:
@luiscuenca Sorry, late to the party here -- does that mean that when a dev passes in "hostURL" it needs to be a complete URL of the format `https://whatever.highfidelity.com/` and NOT just `whatever.highfidelity.com`? I had interpreted from the "`wws://${hostURL}:8001/?token=`" comment you had above that `whatever.highfidelity.com` was the right format. Can you clarify in the comments exactly which format is expected? Thank you!!